hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java

```diff
@@ -201,22 +200,46 @@ protected Map<TokenID, List<NftTransfer>> nftChangesFrom(
             if (receiverAccountId.equals(senderAccountId)) {
                 continue;
             }
-            final var nftTransfer = builder.serialNumber(nftId.serialNumber())
-                    .senderAccountID(senderAccountId)
-                    .receiverAccountID(receiverAccountId)
-                    .build();
+            updateNftChanges(nftId, senderAccountId, receiverAccountId, nftChanges);
+        }
 
-            if (!nftChanges.containsKey(nftId.tokenId())) {
-                nftChanges.put(nftId.tokenId(), new ArrayList<>());
+        for (final var tokenId : writableTokenStore.modifiedTokens()) {
+            final var originalToken = writableTokenStore.getOriginalValue(tokenId);
+            final var modifiedToken = writableTokenStore.get(tokenId);
+            if (originalToken != null
```

Comment:
For self-documenting code could extract this condition to a method so we have,
```
if (bothExistButTreasuryChanged(originalToken, modifiedToken)) {
   ...
```

Reply:
Fixed. Thanks @tinker-michaelj